src/pages/fieldManagement/settings/defaultFieldComponents/relationshipComponents/Type.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { get, omit } from 'lodash-es';
+import { v4 as uuid } from 'uuid';
+
+import InputAdornment from '@material-ui/core/InputAdornment';
+import IconButton from '@material-ui/core/IconButton';
+import NewChildIcon from '@material-ui/icons/AddCircle';
+
+import TextInput from '../../../../../components/inputs/TextInput';
+import DeleteButton from '../../../../../components/DeleteButton';
+import Role from './Role';
+
+function deleteType(relationships, typeGuid) {
+  return omit(relationships, [typeGuid]);
```

Comment:
Doesn't this suffer from the same within-object editing issue that we were seeing on my PR?

Reply:
it says in the lodash docs that omit returns a brand new object: "this method creates an object composed of the own and inherited enumerable property paths of object that are not omitted."